notebooks/contrib-dev/Alcohol_Pharmacokinetics.ipynb

```diff
@@ -0,0 +1,1075 @@
+        "plt.show()\n",
+        "\n",
+        "# Plot C0 and BAC\n",
+        "fig, ax1 = plt.subplots(figsize=(6.4, 4), dpi=100)\n",
+        "ax1.set_xlabel(\"time (hr)\", fontsize=14)\n",
+        "ax1.set_ylabel(\"C0 (g/L)\", color=\"blue\", fontsize=14)\n",
+        "ax1.plot(C[4], C[0], color=\"blue\")\n",
+        "ax1.tick_params(axis=\"y\", labelcolor=\"blue\")\n",
+        "\n",
+        "ax2 = ax1.twinx()\n",
+        "ax2.set_ylabel(\"BAC (%)\", color=\"red\", fontsize=14)\n",
+        "ax2.plot(C[4], C[3], color=\"red\")\n",
+        "ax2.tick_params(axis=\"y\", labelcolor=\"red\")\n",
+        "\n",
+        "plt.title(\"Alcohol Consumption and Elimination Kinetics\", fontsize=16, fontweight=\"bold\")\n",
+        "plt.show()\n",
+        "\n",
+        "# Find the time when BAC falls below the legal limit\n",
+        "legal_limit = 0.08  # 0.08%\n",
+        "\n",
+        "# Find the index where BAC first exceeds the legal limit\n",
+        "above_limit_indices = np.where(C[3] > legal_limit)[0]\n",
+        "\n",
+        "# If there's at least one point where BAC is above the limit, proceed\n",
+        "if above_limit_indices.size > 0:\n",
+        "    # Find the first index where BAC is above the limit\n",
+        "    first_above_index = above_limit_indices[0]\n",
+        "\n",
+        "    # Search from the first_above_index to avoid catching the initial rise\n",
+        "    below_limit_indices = np.where(C[3][first_above_index:] < legal_limit)[0]\n",
+        "\n",
+        "    # Avoid the first time\n",
+        "    if below_limit_indices.size > 0:\n",
+        "        # Add first_above_index to it to get the index in the original array\n",
+        "        first_below_index = below_limit_indices[0] + first_above_index\n",
+        "\n",
+        "        # Find the time\n",
+        "        time_falls_below = C[4][first_below_index]\n",
+        "        print(f\"Time to sober up to below the legal limit: {time_falls_below:.2f} hours\")\n",
+        "    else:\n",
+        "        print(\"BAC does not fall below the legal limit within the time frame.\")\n",
+        "else:\n",
+        "    print(\"BAC never exceeds the legal limit within the time frame.\")\n",
+        "\n",
+        "\n",
+        "### END SOLUTION"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "C3HXfCDa2-M0"
+      },
+      "source": [
+        "Using the plotted kinetics **Discuss in 3-6 sentences** how long it takes for alcohol to enter the blood stream vs how long it takes for the alcohol to break down. How would BAC change over time if the person weighed more/less? If they drink more/less?\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "VK-3NOub6R64"
+      },
+      "source": [
+        "**Discussion:**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## 1D. Example of Forward Euler Method for ODEs\n",
```

Comment:
This contribution is confusing. Is the main goal to practice forward Euler? Or is the main goal to compare the full kinetic and simplified kinetic models? Please make this more clear. If the later, remove the forward Euler and restructure to emphasize comparing different models. Use the same integration scheme as earlier in the notebook.